### PR TITLE
Remove error label when user fixes invalid field

### LINKED ISSUE #1388: Metadata editor leaves field in error state following valid entry
**Describe the issue**
When triggering a required value error in the pipeline runtime editor, the field remains in an error state even after entering data.  This makes it confusing as to which fields still must be addressed.

**To Reproduce**
1. Go to the runtimes editor and enter a name.
2. Click on save and should be presented with required value errors on several fields that require a value
3. Enter "valid" information that should suppress the error state.
4. Can't easily determine which of the remaining error fields still need to be addressed.

**Screenshots or log output**
image

**Expected behavior**
I had expected the field error indicator to be removed once a "valid" entry has been entered.

**Deployment information**
Describe what you've deployed and how:
 - Elyra version: 2.1.0rc2
 - Operating system: macos
 - Installation source: PyPi
 - Deployment type: local installation

### PR DESCRIPTION
**Pull request by marthacryan**
Fixes #1388. Removes the error label when user fixes invalid field. 
 
Developer's Certificate of Origin 1.1

       By making a contribution to this project, I certify that:

       (a) The contribution was created in whole or in part by me and I
           have the right to submit it under the Apache License 2.0; or

       (b) The contribution is based upon previous work that, to the best
           of my knowledge, is covered under an appropriate open source
           license and I have the right under that license to submit that
           work with modifications, whether created in whole or in part
           by me, under the same open source license (unless I am
           permitted to submit under a different license), as indicated
           in the file; or

       (c) The contribution was provided directly to me by some other
           person who certified (a), (b) or (c) and I have not modified
           it.

       (d) I understand and agree that this project and the contribution
           are public and that a record of the contribution (including all
           personal information I submit with it, including my sign-off) is
           maintained indefinitely and may be redistributed consistent with
           this project or the open source license(s) involved.

### CODE REVIEW

**Review by lresende — APPROVED** [review]

**Comment by lresende** [comment]
Tested the metadata editor by entering invalid/valid values and all seem to be showing/removing the error messages based on the correct values.

**Comment by ajbozarth** [comment]
Code looks good to me, I haven't had a chance to test locally though since I'm in the middle of testing 1404

File: packages/metadata-common/src/MetadataEditor.tsx [hunk 1]
```diff
@@ -59,14 +59,26 @@ interface ICodeBlockProps {
...
+  onChange,
+  initialError,
+  label,
+  required
 }) => {
+  const [error, setError] = React.useState(initialError);
+
+  React.useEffect(() => {
+    setError(initialError);
+  }, [initialError]);
```
Comment (bourdakos1) [inline packages/metadata-common/src/MetadataEditor.tsx:80]: this shouldn't be needed

**Review by bourdakos1 — APPROVED** [review]

**Comment by bourdakos1** [comment]
I updated `DropDown`, `TextInput` and `CodeBlock` to be uncontrolled components, because we aren't using react state in the parent component so they can't be real controlled components.

As an uncontrolled components each form control handles its own state, and notifies the parent of any changes via "onChange". The parent component has no control of the form controls state and can only give it an initial value when the control is first mounted.

The `CodeBlock` component relies on `language` from one of the `DropDown` instances, so the editor must be refreshed when `language` changes. I updated `setFormFocus` to check if any of the form controls had focus before giving focus to the `display_name` input. This stops `display_name` from stealing focus as you type a language into `DropDown`

**Comment by lresende** [comment]
@bourdakos1 looks like your change might have caused some component ids/types to change and is affecting the `code-snippet` tests:

image

**Pull request merged**